packages/web/src/pages/edit-coin-details-page/EditCoinDetailsPage.tsx

```diff
@@ -0,0 +1,324 @@
+
+// Helper function to detect platform from URL
+const detectPlatform = (
+  url: string
+): 'x' | 'instagram' | 'tiktok' | 'website' => {
+  const cleanUrl = url.toLowerCase().trim()
+
+  if (cleanUrl.includes('twitter.com') || cleanUrl.includes('x.com')) {
+    return 'x'
+  }
+  if (cleanUrl.includes('instagram.com')) {
+    return 'instagram'
+  }
+  if (cleanUrl.includes('tiktok.com')) {
+    return 'tiktok'
+  }
+
+  return 'website'
+}
+
+// Get platform icon
+const getPlatformIcon = (platform: string) => {
+  switch (platform) {
+    case 'x':
+      return IconX
+    case 'instagram':
+      return IconInstagram
+    case 'tiktok':
+      return IconTikTok
+    case 'website':
+    default:
+      return IconLink
+  }
+}
+
+const SocialLinksSection = () => {
+  const { values, setFieldValue, errors, touched } =
+    useFormikContext<EditCoinDetailsFormValues>()
+
+  const handleAddSocialLink = () => {
+    const newLinks = [...values.socialLinks, '']
+    setFieldValue('socialLinks', newLinks)
+  }
+
+  const handleLinkChange = (
+    index: number,
+    value: ChangeEvent<HTMLInputElement> | string
+  ) => {
+    const newValue = typeof value === 'string' ? value : value.target.value
+    const newLinks = [...values.socialLinks]
+    newLinks[index] = newValue
+    setFieldValue('socialLinks', newLinks)
+  }
+
+  return (
+    <Flex column gap='l' m='xl'>
+      <Flex alignItems='center' gap='s'>
+        <Text variant='title' size='l'>
+          {coinDetailsMessages.editCoinDetails.socialLinks}
+        </Text>
+        <Text variant='body' size='m' color='subdued'>
+          {coinDetailsMessages.editCoinDetails.optional}
+        </Text>
+      </Flex>
+
+      <Flex column gap='l' alignItems='flex-start'>
+        {values.socialLinks.map((link: string, index: number) => {
+          const platform = detectPlatform(link)
+          const PlatformIcon = getPlatformIcon(platform)
+          const fieldError = Array.isArray(errors.socialLinks)
+            ? errors.socialLinks[index]
+            : undefined
+          const fieldTouched = Array.isArray(touched.socialLinks)
+            ? touched.socialLinks[index]
+            : touched.socialLinks
+          const hasError = Boolean(fieldTouched && fieldError)
+
+          return (
+            <TextInput
+              key={index}
+              label={`${coinDetailsMessages.editCoinDetails.socialLink} ${index + 1}`}
+              placeholder={coinDetailsMessages.editCoinDetails.pasteLink}
+              hideLabel
+              value={link}
+              onChange={(value) => handleLinkChange(index, value)}
+              startIcon={PlatformIcon}
+              error={hasError}
+              helperText={hasError ? fieldError : undefined}
+            />
+          )
+        })}
+
+        {/* Add new link button */}
+        {values.socialLinks.length < 4 && (
+          <PlainButton onClick={handleAddSocialLink} iconLeft={IconPlus}>
```

Comment:
TextLink vs PlainButton. Might need plainButton here but wanted to call it out just in case

Reply:
we have a TextLink component. was wondering if we should use that here or if the PlainButton is correct. depends on the designs mostly.